lib/ansible/modules/windows/win_psrepository.ps1

```diff
@@ -1,68 +1,229 @@
+$script_source_location = Get-AnsibleParam -obj $params -name "script_source_location" -aliases "script_source" -type "str"
+$publish_location = Get-AnsibleParam -obj $params -name "publish_location" -aliases "module_publish_location" -type "str"
+$script_publish_location = Get-AnsibleParam -obj $params -name "script_publish_location" -type "str"
 $state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present", "absent"
-$installationpolicy = Get-AnsibleParam -obj $params -name "installation_policy" -type "str" -validateset "trusted", "untrusted"
+$installation_policy = Get-AnsibleParam -obj $params -name "installation_policy" -type "str" -validateset "trusted", "untrusted"
+$resolve_locations = Get-AnsibleParam -obj $params -name "resolve_locations" -type "bool" -default $true
+$force = Get-AnsibleParam -obj $params -name "force" -type "bool" -default $false
 
 $result = @{"changed" = $false}
 
+Function Resolve-PSGetLocation {
+    <#
+        .SYNOPSIS
+        This is a sort of proxy function for the Resolve-Location function that PowerShellGet uses internally.
+
+        .DESCRIPTION
+        It's not exported from the module, and it needs a caller context so wrapping it in a function both convenient
+        and functional depending on how the outer module is executed (where $PSCmdlet may not be set).
+
+        Also wraps the needed to call to other internal function, Get-LocationString, which does some normalizing
+        or [uri] to [string] conversion.
+
+        .NOTES
+        - This must be an advanced function so that $PSCmdlet is populated.
+        - Some context around all the things this tries to do: https://github.com/PowerShell/PowerShellGet/issues/317
+    #>
+    [CmdletBinding()]
+    [OutputType([string])]
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [uri]
+        $Location,
+
+        [Parameter(Mandatory=$true)]
+        [string]
+        $LocationParameterName,
+
+        [Parameter()]
+        [pscredential]
+        $Credential,
+
+        [Parameter()]
+        $Proxy,
+
+        [Parameter()]
+        [pscredential]
+        $ProxyCredential
+    )
+
+    $module = Get-Module -Name PowerShellGet
+
+    $module.Invoke({
+        param(
+            [System.Collections.IDictionary]
+            $Params,
+
+            [System.Management.Automation.PSCmdlet]
+            $CallerPSCmdlet
+        )
+        $Params.Location = Get-LocationString -LocationUri $Params.Location
+        Resolve-Location @Params -CallerPSCmdlet $CallerPSCmdlet
+    }, @($PSBoundParameters,$PSCmdlet))
+}
+
+$repository_params = @{
+    Name = $name
+}
+
+$Repo = Get-PSRepository @repository_params -ErrorAction Ignore
+
+if ($installation_policy) {
+    $repository_params.InstallationPolicy = $installation_policy
+}
+
+# Validate location params are valid URIs and add them to the params
+if ($source_location) {
```

Comment:
You have a lot of common code here, might be best to put in a loop or in your own cmdlet.

Reply:
Ok I'll see what I can do to clean that up